build_for_ultimaker.sh

```diff
@@ -43,13 +43,138 @@ DEB_DIR=`pwd`/debian
 KCONFIG=`pwd`/configs/${BUILDCONFIG}_config
 KERNEL_BUILD=`pwd`/_build_armhf/${BUILDCONFIG}-linux
 
+INITRAMFS_MODULES="sunxi_wdt.ko"
+INITRAMFS_COMPRESSION="${INITRAMFS_COMPRESSION:-.lzo}"
+INITRAMFS_ROOT_GID=${INITRAMFS_ROOT_GID:-0}
+INITRAMFS_ROOT_UID=${INITRAMFS_ROOT_UID:-0}
+INITRAMFS_SOURCE="${INITRAMFS_SOURCE:-initramfs/initramfs.lst}"
+INITRAMFS_IMG="${KERNEL_BUILD}/initramfs.cpio${INITRAMFS_COMPRESSION}"
+
+GEN_INIT_CPIO="${KERNEL_BUILD}/usr/gen_init_cpio"
+GEN_INITRAMFS_LIST="${KERNEL}/scripts/gen_initramfs_list.sh"
+
+BB_PKG="http://ftp.nl.debian.org/debian/pool/main/b/busybox/busybox-static_1.22.0-19+b3_armhf.deb"
+BB_BIN="busybox"
+
+DEPMOD="${DEPMOD:-/sbin/depmod}"
+if [ ! -x "${DEPMOD}" ]; then
+    DEPMOD="busybox depmod"
+    if [ ! -x "${DEPMOD}" ]; then
+        echo "No depmod binary available. Cannot continue."
+        exit 1
+    fi
+fi
+
 # Set the release version if it's not passed to the script
 RELEASE_VERSION=${RELEASE_VERSION:-9999.99.99}
 
 # Initialize repositories
 git submodule init
 git submodule update
 
+busybox_get()
+{
+    local BB_DIR=$(mktemp -d)
+    local BB_AR="${BB_DIR}/busybox-static_armhf.deb"
+    local INITRAMFS_DIR="${1}"
+    local cwd=$(pwd)
+
+    if [ ! -d "${INITRAMFS_DIR}" ]; then
+        echo "No initramfs dir set to download busybox into."
+        exit 1
+    fi
+
+    if [ -z "${BB_DIR}" ]; then
+        echo "Unable to create temporary directory to get busybox."
+        exit 1
+    fi
+
+    wget -q "${BB_PKG}" -O "${BB_AR}"
+    cd "${BB_DIR}" # ar always extracts to the cwd
+    ar -x "${BB_AR}" "data.tar.xz"
+    cd "${cwd}"
+    tar -xf "${BB_DIR}/data.tar.xz" --strip=2 -C "${BB_DIR}" "./bin/${BB_BIN}"
+    mv "${BB_DIR}/${BB_BIN}" "${INITRAMFS_DIR}"
+    rm -r "${BB_DIR}"
```

Comment:
Not really needed it is tempfs

Reply:
well for one, it's nice to clean up after oneself, secondly, tmpfs only gets cleaned up after a reboot (and systemd occasionally cleans it up) so if you run this build script, say 10 days on a day, you'll have 10 stale directories lingering. Not very nice and wasteful of diskspace :)